src/Text/Text.js

```diff
@@ -1,19 +1,68 @@
 import React, { forwardRef } from 'react'
 import classNames from 'classnames'
 
-// this component is not fully fleshed out, it's just here to add some typography
-function Text({ className, as, children }, ref) {
-  const Component = as ?? 'span'
+// this is the variant approach to text
+function Text({ as, variant, children }, ref) {
+  const Component = as ?? getHTMLElement(variant)
+
+  var className
+  switch (variant) {
+    case 'h1':
+      className = 'font-display font-bold text-6xl'
+      break
+    case 'h2':
+      className = 'font-display font-semibold text-5xl'
+      break
+    case 'h3':
+      className = 'font-display font-medium text-4xl'
+      break
+    case 'h4':
+      className = 'font-display font-normal text-3xl'
+      break
+    case 'h5':
+      className = 'font-display font-light text-2xl'
+      break
+    case 'h6':
+      className = 'font-display font-thin text-xl'
+      break
+    case 'bodyLarge':
+      className = 'font-body font-medium text-lg'
+      break
+    case 'bodyBase':
+      className = 'font-body font-normal text-base'
+      break
+    case 'bodySmall':
+      className = 'font-body font-light text-sm'
+      break
+    case 'label':
+      className = 'font-body font-normal text-xs'
+      break
+    default:
+      className = ''
+  }
 
   return (
     <Component
       ref={ref}
-      className={classNames('font-display text-base', className)}
+      className={classNames('text-gray-pink-400', className)}
     >
       {children}
     </Component>
   )
 }
 
+const getHTMLElement = variant => {
+  if (
```

Comment:
Whenever I have a situation like this, I like to write it as `['h1','h2','h3','h4','h5','h6',].includes(variant)`. You definitely don't have too, I just find it a little easier.

Reply:
Cool. I just copied from what we are doing in Calumet, so clearly you didn't write that code either, haha

Reply:
Haha ah, I was a sillier person back then, who liked to make things more difficult for myself